hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenRejectHandler.java

```diff
@@ -118,9 +158,196 @@ public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckExcept
 
     @Override
     public void handle(@NonNull final HandleContext context) throws HandleException {
+        requireNonNull(context);
         final var op = context.body().tokenRejectOrThrow();
+        final var rejectingAccountID = op.ownerOrElse(context.payer());
         final var rejections = op.rejections();
-        // Todo: Implement the logic for token rejection
+
+        final var hederaConfig = context.configuration().getConfigData(HederaConfig.class);
+        final var ledgerConfig = context.configuration().getConfigData(LedgerConfig.class);
+        final var tokensConfig = context.configuration().getConfigData(TokensConfig.class);
+        validateTrue(tokensConfig.tokenRejectEnabled(), NOT_SUPPORTED);
+        validateTrue(rejections.size() <= ledgerConfig.tokenRejectsMaxLen(), INVALID_TRANSACTION_BODY);
```

Comment:
We can have `REJECT_LIST_SIZE_LIMIT_EXCEEDED` like `TRANSFER_LIST_SIZE_LIMIT_EXCEEDED`

Reply:
We could add such a status.
 Do you think we can also add statuses for other cases, or do you think the others are fine? Asking so we could open one PR for all of them in the protobuf repo and ask for feedback.

Reply:
I can only think of this 🤔